backend/tournesol/management/commands/create_dataset.py

```diff
@@ -0,0 +1,110 @@
+"""
+Create and save a public dataset archive on the disk.
+"""
+import zipfile
+from io import StringIO
+from os.path import getctime
+from pathlib import Path
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+from tournesol.lib.public_dataset import (
+    write_comparisons_file,
+    write_individual_criteria_scores_file,
+    write_users_file,
+)
+from tournesol.models.poll import Poll
+
+
+class Command(BaseCommand):
+    help = "Create and save a public dataset archive on the disk, then delete old datasets."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--keep-only",
+            type=int,
+            default=10,
+            help="The number of archives to keep on the disk (default 10)."
+            " The oldest archives will be deleted. No effect when --keep-all is set.",
+        )
+        parser.add_argument(
+            "--keep-all",
+            action="store_true",
+            default=False,
+            help="Do not remove old archives from the disk."
+            " Ignore the value defined by --keep-only.",
+        )
+
+    def handle(self, *args, **options):
+        """
+        Create a public dataset archive with fresh data retrieved from the
+        database.
+
+        The archive is created on the disk at:
+            `MEDIA_ROOT/APP_TOURNESOL["DATASETS_BUILD_DIR"]`
+        """
+        self.stdout.write(f"start command: {__name__}")
+
+        # Display the configuration if more verbosity is asked.
+        if options.get("verbosity", 1) > 1:
+            self.stdout.write(f"MEDIA_ROOT: {settings.MEDIA_ROOT}")
+            self.stdout.write(
+                "APP_TOURNESOL['DATASETS_BUILD_DIR']:"
+                f" {settings.APP_TOURNESOL['DATASETS_BUILD_DIR']}"
+            )
+
+        dataset_base_name = settings.APP_TOURNESOL["DATASET_BASE_NAME"]
+        datasets_build_dir = Path(settings.MEDIA_ROOT).joinpath(
+            settings.APP_TOURNESOL["DATASETS_BUILD_DIR"]
+        )
+        datasets_build_dir.mkdir(parents=True, exist_ok=True)
+
+        # Only the default poll is exported for the moment.
+        poll_name = Poll.default_poll().name
+
+        archive_name = f"{dataset_base_name}{timezone.now().strftime('%Y%m%d')}"
+        archive_root = datasets_build_dir.joinpath(archive_name)
+        readme_path = "tournesol/resources/export_readme.txt"
```

Comment:
Maybe we could also use a path object here

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1359/commits/daf26d22fd8ad51e604ef9995244a1970525d56f